freedom/reco/crs_reco.py

```diff
@@ -32,43 +35,69 @@
 from freedom.reco import summary_df
 from freedom.reco import prefit
 
+#load time residual spline function
+from scipy.interpolate import UnivariateSpline
+from freedom.utils.time_residual import TimeResidual
+with open(pkg_resources.resource_filename('freedom', 'resources/prior/oscNext_time_residual_prior.pkl'), 'rb') as F:
+    ftr = pickle.load(F)
+ftr = UnivariateSpline._from_tck(ftr)
+ftr.ext = 1
 
-def get_batch_closure(client, event, out_of_bounds):
+
+def get_batch_closure(clients, event, out_of_bounds, Tprior=True):
     """returns LLH batch eval closure for this event. The closure is a function of one argument: the hypotheses to evaluate"""
     hit_data = event["hit_data"]
     evt_data = event["evt_data"]
-
+    
     def eval_llh(params):
-        llhs = np.atleast_1d(client.eval_llh(hit_data, evt_data, params))
+        llhs = 0
+        for i in range(len(clients)):
+            llhs += np.atleast_1d(clients[i].eval_llh(hit_data[i], evt_data[i], params))
+            
+            if Tprior:
+                tresiduals = TimeResidual(np.tile(hit_data[i], (len(llhs),1)).astype(np.float32), 
+                                          np.repeat(params, len(hit_data[i]), axis=0).astype(np.float32))
+                tresiduals = np.log(np.clip(ftr(tresiduals), 1e-10, 1))
```

Comment:
these are now not residuals anymore, but log_prior value, right?

Reply:
ftr (which will be renamed) turns the time residuals into probabilities. So here we have log(p) which is then just added to the LLH. Maybe I should use a new variable for the output of line 60?

Reply:
yep, that would be better